execution/executor-benchmark/src/native/native_transaction.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_types::{
+    account_address::AccountAddress,
+    transaction::signature_verified_transaction::SignatureVerifiedTransaction,
+};
+
+#[derive(Debug)]
+pub enum NativeTransaction {
+    Nop {
+        sender: AccountAddress,
+        sequence_number: u64,
+    },
+    FaTransfer {
+        sender: AccountAddress,
+        sequence_number: u64,
+        recipient: AccountAddress,
+        amount: u64,
+    },
+    Transfer {
+        sender: AccountAddress,
+        sequence_number: u64,
+        recipient: AccountAddress,
+        amount: u64,
+        fail_on_recipient_account_existing: bool,
+        fail_on_recipient_account_missing: bool,
+    },
+    BatchTransfer {
+        sender: AccountAddress,
+        sequence_number: u64,
+        recipients: Vec<AccountAddress>,
+        amounts: Vec<u64>,
+        fail_on_recipient_account_existing: bool,
+        fail_on_recipient_account_missing: bool,
+    },
+}
+
+impl NativeTransaction {
```

Comment:
In my remote sharding experiments, native vm was not able to handle this one txn in the beginning. You have changed the code since then, but it might worthwhile to run an "unsharded execution" in the executor-benchmark 
```
2024-11-16T00:33:08.305178Z [main] INFO storage/aptosdb/src/state_store/mod.rs:451 Initializing BufferedState. {"latest_snapshot_version":null,"num_transactions":0}
2024-11-16T00:33:08.305256Z [main] INFO storage/aptosdb/src/state_store/mod.rs:547 StateStore initialization finished. {"latest_in_memory_root_hash":"5350415253455f4d45524b4c455f504c414345484f4c4445525f484153480000","latest_in_memory_version":null,"latest_snapshot_root_hash":"5350415253455f4d45524b4c455f504c414345484f4c4445525f484153480000","latest_snapshot_version":null}
2024-11-16T00:33:08.305735Z [main] INFO aptos-move/aptos-vm/src/aptos_vm.rs:2557 Executing block, transaction count: 1 {"name":"miscellaneous","txn_idx":0}
```